pkg/controller/installation/windows_controller.go

```diff
@@ -0,0 +1,455 @@
+	reqLogger := logw.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling Installation.operator.tigera.io")
+
+	// Get the installation object if it exists so that we can save the original
+	// status before we merge/fill that object with other values.
+	instance := &operatorv1.Installation{}
+	if err := r.client.Get(ctx, utils.DefaultInstanceKey, instance); err != nil {
+		if apierrors.IsNotFound(err) {
+			reqLogger.Info("Installation config not found")
+			r.status.OnCRNotFound()
+			return reconcile.Result{}, nil
+		}
+		reqLogger.Error(err, "An error occurred when querying the Installation resource")
+		return reconcile.Result{}, err
+	}
+
+	// Don't render calico-node-windows if it's disabled in the installation
+	if !common.WindowsEnabled(instance.Spec) {
+		reqLogger.Info("Calico Windows daemonset is disabled in the operator installation")
+		return reconcile.Result{}, nil
+	}
+
+	// Mark CR found so we can report converter problems via tigerastatus
+	r.status.OnCRFound()
+	// FIXME: add logic to merge Installation status metadata
+
+	// Changes for updating Installation status conditions.
+	if request.Name == InstallationName && request.Namespace == "" {
+		ts := &operatorv1.TigeraStatus{}
+		err := r.client.Get(ctx, types.NamespacedName{Name: InstallationName}, ts)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		instance.Status.Conditions = status.UpdateStatusCondition(instance.Status.Conditions, ts.Status.Conditions)
+		if err := r.client.Status().Update(ctx, instance); err != nil {
+			logw.WithValues("reason", err).Info("Failed to create Installation status conditions.")
+			return reconcile.Result{}, err
+		}
+	}
+
+	instanceStatus := instance.Status
+
+	reqLogger.V(2).Info("Loaded config", "config", instance)
+
+	// Validate the configuration.
+	if err := validateCustomResource(instance); err != nil {
+		r.status.SetDegraded(operatorv1.InvalidConfigurationError, "Invalid Installation provided", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	// update Installation with 'overlay'
+	overlay := operatorv1.Installation{}
+	if err := r.client.Get(ctx, utils.OverlayInstanceKey, &overlay); err != nil {
+		if !apierrors.IsNotFound(err) {
+			reqLogger.Error(err, "An error occurred when querying the 'overlay' Installation resource")
+			return reconcile.Result{}, err
+		}
+		reqLogger.V(5).Info("no 'overlay' installation found")
+	} else {
+		instance.Spec = utils.OverrideInstallationSpec(instance.Spec, overlay.Spec)
+		reqLogger.V(2).Info("loaded final computed config", "config", instance)
+
+		// Validate the configuration.
+		if err := validateCustomResource(instance); err != nil {
+			r.status.SetDegraded(operatorv1.InvalidConfigurationError, "Invalid computed config", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+	}
+
+	// A status is needed at this point for operator scorecard tests.
```

Comment:
This comment should say "We rely on the core controller for defaulting, so wait until it has done so before continuing"